pkg/workflowengine/interfaces/registry.go

```diff
@@ -0,0 +1,12 @@
+package interfaces
+
+// FlyteK8sWorkflowExecutorRegistry is a singleton provider of a K8sWorkflowExecutor implementation to use for
+// creating and deleting Flyte workflow CRD objects.
+type FlyteK8sWorkflowExecutorRegistry interface {
+	// Registers a new K8sWorkflowExecutor to handle creating and aborting Flyte workflow executions.
+	Register(executor K8sWorkflowExecutor)
+	// Registers the default K8sWorkflowExecutor to handle creating and aborting Flyte workflow executions.
+	RegisterDefault(executor K8sWorkflowExecutor)
```

Comment:
Will there be at a time default and non-default workflow executors in the system. Because i dont see reference to Register()

Reply:
mind checking the offline thread for an explanation? :)

Reply:
Thanks for clarifying this offline